src/Amadeus/Client/Struct/Service/IntegratedPricing.php

```diff
@@ -155,6 +161,37 @@ protected function loadPricingOptions($options)
 
         return $priceOptions;
     }
+    
+    /**
+     * @param FareBasis[] $pricingsFareBasis
+     * @return PricingOptionGroup[]
+     */
+    protected static function makePricingOptionFareBasisOverride($pricingsFareBasis)
+    {
+    	$opt = [];
+    
+    	if ($pricingsFareBasis !== null) {
+    		foreach ($pricingsFareBasis as $pricingFareBasis) {
+    			$po = new PricingOptionGroup(PricingOptionKey::OPTION_FARE_BASIS_SIMPLE_OVERRIDE);
+    
+    			//Support for legacy fareBasisPrimaryCode to be removed when breaking BC:
+    			$po->optionDetail = new OptionDetail(
```

Comment:
Where is this code coming from? You shouldn't be adding new code with support for deprecated features. If this is a new feature, it doesn't need to support deprecated behaviour.

Reply:
I have change this function now, We need farebasis option along with passenger and segment association.